sql/create.go

```diff
@@ -142,6 +142,8 @@ func (p *planner) CreateIndex(n *parser.CreateIndex) (planNode, error) {
 		return nil, err
 	}
 
+	// Mark transaction as operating on the system DB.
+	p.txn.SetSystemDBTrigger()
```

Comment:
can we be smarter about this? i.e. when the txn is running the batch it should be able to tell if operators on the system DB or not, similarly to the current `haveTxnWrite/haveEndTxn` handling